components/raftstore/src/store/peer.rs

```diff
@@ -158,6 +159,36 @@ pub struct CheckTickResult {
     up_to_date: bool,
 }
 
+pub struct BatchRaftCmdRequest {
+    pub request: Option<RaftCmdRequest>,
+    pub batch_size: u32,
+    pub callbacks: Vec<(Callback<RocksEngine>, usize)>,
+}
+
+impl BatchRaftCmdRequest {
+    fn new() -> BatchRaftCmdRequest {
+        BatchRaftCmdRequest {
+            request: None,
+            batch_size: 0,
+            callbacks: vec![],
+        }
+    }
+
+    fn push(&mut self, mut req: RaftCmdRequest, req_size: u32, cb: Callback<RocksEngine>) {
+        let req_num = req.get_requests().len();
+        if let Some(batch_req) = self.request.as_mut() {
+            let requests: Vec<_> = req.take_requests().into();
+            for q in requests {
+                batch_req.mut_requests().push(q);
```

Comment:
Have you considered admin request and the context in request header? https://github.com/pingcap/kvproto/blob/master/proto/raft_cmdpb.proto#L301

Reply:
No. A admin request would not be merged with other request. Please see code `peer.rs#1762`

Reply:
It is easy to be misused.

Reply:
I will add one `assert` for this check.

Reply:
And give some explicit comments.